src/solvers/refinement/string_refinement.cpp

```diff
@@ -1011,31 +1075,71 @@ bool string_refinementt::check_axioms()
     }
   }
 
-  // tells whether one of the not_contains constraint can be violated
-  bool violated_not_contains=false;
+  // Maps from indexes of violated not_contains axiom to a witness of violation
+  std::map<size_t, exprt> violated_not;
+
   debug() << "there are " << not_contains_axioms.size()
           << " not_contains axioms" << eom;
-
-  for(auto nc_axiom : not_contains_axioms)
+  for(size_t i=0; i<not_contains_axioms.size(); i++)
```

Comment:
It's used slightly below to keep track of how the constraint is violated by storing it in `violated_not_contains` for debugging purposes and there is code that shows it will be used in the future for counter examples.